### PR TITLE
Rewrite license code to avoid async*

### LINKED ISSUE #95043: Refactor license logic in the framework to use isolates now that they are cheaper
This should also help avoid using `async*`, which is currently difficult to refactor away from because of the lack of isoalte processing.

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/95043

This takes another 5kb off hello_world.

This is meant to be a compatible refactor that will show improvements on benchmarks.

### CODE REVIEW

**Review by a14n — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/foundation/licenses.dart [hunk 1]
```diff
@@ -98,7 +103,10 @@ enum _LicenseEntryWithLineBreaksParserState {
 /// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
 /// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 /// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-/// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.''');
+/// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.'''));
+///       },
+///     );
+///     return controller.stream;
```
[the remark this answers is not in the document]
Comment (dnfield) [reply packages/flutter/lib/src/foundation/licenses.dart:109]: Ooo even better.

File: packages/flutter/lib/src/foundation/licenses.dart [hunk 2]
```diff
@@ -104,10 +101,7 @@ enum _LicenseEntryWithLineBreaksParserState {
 /// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 /// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 /// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.'''));
-///       },
-///     );
-///     return controller.stream;
-///   });
+///   ));
```
[the remark this answers is not in the document]
Comment (dnfield) [reply packages/flutter/lib/src/foundation/licenses.dart:104]: Yes, you are right.

[the remark this answers is not in the document]
Comment (dnfield) [reply packages/flutter/lib/src/foundation/licenses.dart:104]: There was another bug, I put the type parameter on the wrong part. But thankfully there's analysis to catch this now

**Pull request merged**